machine-tests/src/test/java/com/dylibso/chicory/testing/MachinesTest.java

```diff
@@ -204,9 +206,13 @@ public void shouldUseMachineCallOnlyForExport() throws Exception {
                         .withMachineFactory(
                                 (inst) -> {
                                     var machine = Wat2WasmMachineFactory.create(inst);
-                                    return (funcId, args) -> {
-                                        assertEquals(startFunctionIndex.get(), funcId);
-                                        return machine.call(funcId, args);
+                                    return new Machine() {
```

Comment:
This change to use an anonymous class shouldn't be needed anymore. We could make `Machine` have `@FunctionalInterface`.